cmd/thanos/receive.go

```diff
@@ -278,6 +284,7 @@ func runReceive(
 			s := newStoreGRPCServer(logger, reg, tracer, tsdbStore, opts)
 
 			level.Info(logger).Log("msg", "listening for StoreAPI gRPC", "address", grpcBindAddr)
+			statusProber.SetReady()
```

Comment:
I think the receiver should probably not be ready until the TSDB is ready? Also not sure about the hashring and the receive interface is also not guarantied to be up at this point. 

Maybe this will require some more complex condition for the ready state :thinking:

Reply:
@FUSAKLA For the TSDB, it's ready at this stage if you check line `270`. It runs after `TSDB` is open.
For receive interface, I thought if something goes south it'll change liveness state so, readiness won't be needed.

I guess I need to double-check the hashring readiness.

I'll have another look at it.